src/Taylor1.jl

```diff
@@ -666,3 +669,53 @@ function deriv{T<:Number}(a::Taylor1{T}, n::Int=1)
     @assert a.order >= n >= 0
     factorial( widen(n) ) * a.coeffs[n+1] :: T
 end
+
+
+"""
+    A_mul_B!(y, a, b)
+
+Multiply a*b and save the result in y.  Extends y to fit the results if necessary.
+"""
+function A_mul_B!{T<:Number}(y::Vector{Taylor1{T}},a::Matrix{T},b::Vector{Taylor1{T}})
+
+    # determine the maximal order of b
+    order = maximum([b1.order for b1 in b])
+
+    n,k = size(a)
+
+    # extend y to fit all the coefficients and zero the
```

Comment:
I completely agree that it doesn't make sense to initialize `y` in `A_mul_B!` and that the convention is to assume that `y` is ready to be written on.  But this convention relies on an implementation of a function `similar`, which `TaylorSeries` doesn't have yet.  Once it's there you could safely remove the lines you mentioned but I would leave them for now as a temporary measure.

As for the improvement, feel free to merge this PR and implement the correction yourself.  Although I have to admit I don't quite get what you have in mind.  Isn't it how it already works?  Calling `Y=A*B` in my version is roughly equivalent to `B=A_mul_B!(similar(B),A,B)`.